### PR TITLE
Fix clashing ticket ids

### LINKED ISSUE #3702: Bug: Freshdesk tickets may match SuSy ticket ids
**Describe the bug**

Our current filtering for cases belonging to a ticket is oftentimes using the `case.tickets` column, which contains a comma separated list of ticket ids. There are instances where, provided a ticket_id, we just check if the column value contains the ticket_id. This is error prone since the freshdesk ticket ids may very well contain a SuSy ticket id just because they are longer (e.g. 1234567 contains 123456). 

**To Reproduce**
1. Create a case with a Freshdesk 7 number ticket id
2. Use the filtering method using a 6 number ticket id and see if the case created in 1. is returned.

**Expected behavior**

**Additional context**

### PR DESCRIPTION
**Pull request by islean**
## Description

This PR reworks filtering on ticket ids to go via the Order table instead of the tickets column found in the Case table. Closes #3702 

### Added

-

### Changed

- Filtering on ticket id is done in the Order table

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b fix-ticket-clashes -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/store/crud/read.py [hunk 1]
```diff
@@ -313,25 +313,26 @@ def filter_cases_with_samples(self, case_ids: list[str]) -> list[str]:
 
     def get_cases_by_ticket_id(self, ticket_id: str) -> list[Case]:
```
Comment (islean) [inline cg/store/crud/read.py:314]: Would we also want to change the type of the ticket_id to int in this file?

Comment (diitaz93) [reply cg/store/crud/read.py:314]: I see beneficial to have ticket id as int in all our codebase but that seems out of scope

**Review by diitaz93 — APPROVED** [review]
Looks good 👍

File: cg/store/base.py [hunk 2]
```diff
@@ -74,6 +74,12 @@ def _get_join_sample_family_query(self) -> Query:
         """Return a join sample case relationship query."""
         return self._get_query(table=Sample).join(Case.links).join(CaseSample.sample)
 
+    def _get_join_sample_case_order_query(self) -> Query:
+        """Return a join sample case relationship query."""
+        return (
+            self._get_query(table=Sample).join(Case.links).join(CaseSample.sample).join(Case.orders)
+        )
```
[the remark this answers is not in the document]
Comment (islean) [reply cg/store/base.py:81]: It probably will but is it beneficial? To me the current wording is clearer

File: cg/store/crud/read.py (hunk 1 above)
Comment (islean) [reply cg/store/crud/read.py:314]: Yeah, out of scope but it should be possible once everything uses the Order table to get the ticket_id. I could try to see what would happen if the input is changed to int here though

Comment (islean) [reply cg/store/crud/read.py:314]: Seems to impact CLI commands and service instantiations. I will consider out of scope

**Comment by islean** [comment]
Tested:

Ran `cg decompress ticket <ticket_id>` and it successfully chose the relevant sample

**Comment by islean** [comment]
Deployed to Hasta:

```
done.
Log deploy... done.
cg, version 63.5.13
remote: Enumerating objects: 11, done.
remote: Counting objects: 100% (11/11), done.
remote: Compressing objects: 100% (6/6), done.
remote: Total 11 (delta 4), reused 9 (delta 4), pack-reused 0 (from 0)
Unpacking objects: 100% (11/11), done.
From https://github.com/Clinical-Genomics/cg
   d6422d3..af54aaa  master     -> origin/master
 * [new tag]         v63.5.13   -> v63.5.13
Already on 'master'
Your branch is behind 'origin/master' by 2 commits, and can be fast-forwarded.
  (use "git pull" to update your local branch)
Updating d6422d3..af54aaa
Fast-forward
 .bumpversion.cfg                                 |  2 +-
 cg/__init__.py                                   |  2 +-
 cg/store/base.py                                 | 13 ++++++++++++-
 cg/store/crud/read.py                            | 27 ++++++++++++++-------------
 cg/store/filters/status_case_filters.py          |  6 ------
 pyproject.toml                                   |  2 +-
 tests/store/filters/test_status_cases_filters.py | 31 -------------------------------
 tests/store/filters/test_status_order_filters.py | 51 +++++++++++++++++++++++++++++++++++++++++++++++++++
 tests/store_helpers.py                           |  7 ++++++-
 9 files changed, 86 insertions(+), 55 deletions(-)
 create mode 100644 tests/store/filters/test_status_order_filters.py
/home/isak.ohlsson
INFO  [alembic.runtime.migration] Context impl MySQLImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
[isak.ohlsson@hasta:~] [P_base] $ 
```

**Pull request merged**